File: D_Inversion_Graph.cpp
#include <iostream>
#include <vector>

int main() {
    int t;
    std::cin >> t;

    while (t--) {
        int n;
        std::cin >> n;
        std::vector<int> perms(n);

        for (int i = 0; i < n; i++) {
            std::cin >> perms[i];
        }

        int node = 0;
        int components = 0;
        for (int i = 0; i < n; i++) {
            node = std::max(node, perms[i]);
            if (node <= i + 1) {
                components++;
            }
        }

        std::cout << components << std::endl;
    }

    return 0;
}

File: D_Inversion_Graph.py
def main():
    for _ in range(int(input())):
        n = int(input())
        perms = list(map(int, input().split()))

        node = components = 0
        for i in range(n):
            node = max(node, perms[i])
            if node <= i + 1:
                components += 1
        
        print(components)

if __name__ == '__main__':
    main()